docs/self-managed/platform-deployment/helm-kubernetes/upgrade.md

```diff
@@ -0,0 +1,154 @@
+---
+id: upgrade
+title: "Upgrading Camunda Platform 8 Helm deployment"
+sidebar_label: "Upgrade"
+---
+
+To upgrade to a more recent version of the Camunda Platform Helm charts, there are certain things you need to keep in mind.
+
+## General upgrade instructions
+
+### Upgrading where Identity disabled
+
+Normally for a Helm upgrade, you run the [Helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) command. If you have disabled Camunda Identity component and the related authentication mechanism, you should be able to easily do an upgrade as following:
+
+```sh
+helm upgrade <RELEASE_NAME>
+```
+
+However, if Camunda Identity component is enabled (which is the default), the upgrade path is a bit more complex than just running `helm upgrade`. Read the next section to familiarize yourself with the upgrade process.
+
+### Upgrading where Identity enabled
+
+If you have installed the Camunda Platform 8 Helm charts before with default values, this means Identity and the related authentication mechanism are enabled. For authentication, the Helm charts generate for each web app the secrets randomly if not specified on installation. If you just run `helm upgrade` to upgrade to a newer chart version, you likely will see the following return:
+
+```shell
+helm upgrade camunda-platform-test camunda/camunda-platform
+
+Error: UPGRADE FAILED: execution error at (camunda-platform/charts/identity/templates/tasklist-secret.yaml:10:22):
+PASSWORDS ERROR: You must provide your current passwords when upgrading the release.
+                 Note that even after reinstallation, old credentials may be needed as they may be kept in persistent volume claims.
+                 Further information can be obtained at https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases
+
+    'global.identity.auth.tasklist.existingSecret' must not be empty, please add '--set global.identity.auth.tasklist.existingSecret=$TASKLIST_SECRET' to the command. To get the current value:
+
+        export TASKLIST_SECRET=$(kubectl get secret --namespace "zell-c8-optimize" "camunda-platform-test-tasklist-identity-secret" -o jsonpath="{.data.tasklist-secret}" | base64 --decode)
+```
```

Comment:
Is the spacing here intentional? Particularly in lines 33-35?

And I'm assuming the first line (L26) is the command you run before seeing this error. Is this also included (echoed) in the return? If this is the command the user enters, let's indicate it with a `$`.

Reply:
I admit it looks weird :sweat_smile: 
but the whole section is the output of the first line:
```
helm upgrade camunda-platform-test camunda/camunda-platform
```

I've split it to make it more clear which is the command and which is the output.